electron/index.ts

```diff
@@ -35,6 +35,12 @@ let mainWindow: Electron.BrowserWindow|null;
 
 let tray: Tray;
 let isAppQuitting = false;
+// Default to English strings in case we fail to retrieve them from the renderer process.
```

Comment:
So, this works but I some drawbacks:
- you're duplicating the strings here (as in the rest of the system, let's hope we would catch this before release)
- relatively complex structure passing back and forth between the main and renderer processes

I think a simpler solution would involve synchronous requests. They're safe for functions which are guaranteed to be fast:
https://electronjs.org/docs/api/ipc-main

You could have a function in this file - `localise(key:string) : string` - which returns the result of a synchronous message to the renderer. No maps, and each function is easier to test. WDYT?
https://electronjs.org/docs/api/ipc-main

Reply:
I'll make the l10n function synchronous (thought IpcMain was only synchronous, thanks!). I can remove the duplicate fallback strings, though if l10n fails we would also fail to construct the system tray.
I don't see passing of an string object as too complex of a structure since the Electron's IPC is built to support it. A single batch operation is more efficient that individual l10n requests, but I'm happy to change it if you think it's cleaner.

Reply:
Suggestions:
- if the call fails, fall back to the key name (like the rest of the client)
- passing a string is fine, of course, i meant passing an array of strings - my vote would be a function that translates one string at a time

Reply:
Synchronous IPC is only supported from the renderer process to the main process. As discussed offline, I'll leave the batch call.